addons/sale_stock/tests/test_anglosaxon_account.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+from odoo.addons.account.tests.account_test_savepoint import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+@tagged('post_install', '-at_install')
+class TestAngloSaxonAccounting(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company_data['company'].write({
+            'anglo_saxon_accounting': True
+        })
+        cls.company_data_2['company'].write({
+            'anglo_saxon_accounting': True
+        })
+        product_category = cls.env['product.category'].create({
+            'name': 'a random storable product category',
+            'property_cost_method': 'fifo',
+            'property_valuation': 'real_time',
+        })
+        cls.storable_product = cls.env['product.product'].create({
+            'name': 'storable product a',
+            'type': 'product',
+            'categ_id': product_category.id,
+        })
+        # Those values are company dependent and need to be explicitly set for both companies
+        cls.env.user.company_id = cls.company_data_2['company'].id
+        product_category.write({
+            'property_cost_method': 'fifo',
+            'property_valuation': 'real_time',
+        })
+        cls.env.user.company_id = cls.company_data['company'].id
+
+
+    def test_cogs_should_use_price_from_the_right_company(self):
+        """
+        Reproduce the flow of creating an invoice from a sale order with a company B
+        and posting the invoice with both companies selected and company A as the main.
+        """
+        company_a_data = self.company_data
+        company_b_data = self.company_data_2
+        product = self.storable_product
+
+        # set different cost price for the same product in the 2 companies
+        self.env.user.company_id = company_a_data['company'].id
+        company_a_standard_price = 20.0
+        product.write({'standard_price': company_a_standard_price})
```

Comment:
to write a single value, just do an assignation: `product.standard_price = company_a_standard_price`